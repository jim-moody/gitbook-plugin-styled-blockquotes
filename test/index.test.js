var tester = require('gitbook-tester')
var pluginDir = require('path').join(__dirname, '..')
const examples = require('./examples')

const setupBook = content => {
  return tester
    .builder()
    .withContent(content)
    .withLocalPlugin(pluginDir)
    .create()
}
describe('Rendering blockquotes with and without styling', () => {
  jest.setTimeout(20000)

  it('should return an unchanged blockquote when there is no style defined', done => {
    setupBook(examples.none)
      .then(function (results) {
        expect(results[0].content).toMatchSnapshot()
      })
      .then(done)
      .done()
  })
  test('should return an unchanged blockquote when the defined style is invalid', done => {
    setupBook(examples.invalid)
      .then(results => {
        expect(results[0].content).toMatchSnapshot()
      })
      .then(done)
      .done()
  })
  test('should return correctly rendered html when the defined style is info', done => {
    setupBook(examples.info)
      .then(results => {
        expect(results[0].content).toMatchSnapshot()
      })
      .then(done)
      .done()
  })
  test('should return correctly rendered html when the defined style is info with icon', done => {
    setupBook(examples.infoIcon)
      .then(results => {
        expect(results[0].content).toMatchSnapshot()
      })
      .then(done)
      .done()
  })
  test('should return correctly rendered html when the defined style is tip', done => {
    setupBook(examples.tip)
      .then(results => {
        expect(results[0].content).toMatchSnapshot()
      })
      .then(done)
      .done()
  })
  test('should return correctly rendered html when the defined style is danger', done => {
    setupBook(examples.danger)
      .then(results => {
        expect(results[0].content).toMatchSnapshot()
      })
      .then(done)
      .done()
  })
  test('should return correctly rendered html when the defined style is success', done => {
    setupBook(examples.success)
      .then(results => {
        expect(results[0].content).toMatchSnapshot()
      })
      .then(done)
      .done()
  })
})
