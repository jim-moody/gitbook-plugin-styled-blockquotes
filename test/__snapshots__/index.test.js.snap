// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering blockquotes with and without styling should return an unchanged blockquote when the defined style is invalid 1`] = `
"<blockquote>
<p><strong>warning</strong>
content</p>
</blockquote>"
`;

exports[`Rendering blockquotes with and without styling should return an unchanged blockquote when there is no style defined 1`] = `
"<blockquote>
<p>content</p>
</blockquote>"
`;

exports[`Rendering blockquotes with and without styling should return correctly rendered html when the defined style is danger 1`] = `
"<div class=\\"alert alert-danger hints-alert\\"><div class=\\"hints-icon\\"><i class=\\"fa \\"></i></div><div class=\\"hints-container\\">
<p>
content</p>
</div></div>"
`;

exports[`Rendering blockquotes with and without styling should return correctly rendered html when the defined style is info 1`] = `
"<div class=\\"alert alert-info hints-alert\\"><div class=\\"hints-icon\\"><i class=\\"fa \\"></i></div><div class=\\"hints-container\\">
<p> 
content</p>
</div></div>"
`;

exports[`Rendering blockquotes with and without styling should return correctly rendered html when the defined style is info with icon 1`] = `
"<div class=\\"alert alert-info hints-alert\\"><div class=\\"hints-icon\\"><i class=\\"fa fa fa-info-circle\\"></i></div><div class=\\"hints-container\\">
<p>
content</p>
</div></div>"
`;

exports[`Rendering blockquotes with and without styling should return correctly rendered html when the defined style is success 1`] = `
"<div class=\\"alert alert-success hints-alert\\"><div class=\\"hints-icon\\"><i class=\\"fa \\"></i></div><div class=\\"hints-container\\">
<p>
content</p>
</div></div>"
`;

exports[`Rendering blockquotes with and without styling should return correctly rendered html when the defined style is tip 1`] = `
"<div class=\\"alert alert-warning hints-alert\\"><div class=\\"hints-icon\\"><i class=\\"fa \\"></i></div><div class=\\"hints-container\\">
<p>
content</p>
</div></div>"
`;
